libs/pages/settings/src/lib/feature/page-organization-github-repository-access-feature/page-organization-github-repository-access-feature.tsx

```diff
@@ -0,0 +1,114 @@
+import PageOrganizationGithubRepositoryAccess from '../../ui/page-organization-github-repository-access/page-organization-github-repository-access'
+
+export function PageOrganizationGithubRepositoryAccessFeature() {
+  useDocumentTitle('Github Repository Access - Organization settings')
+  const { organizationId = '' } = useParams()
+
+  const githubConnectUrl = `https://github.com/apps/qovery-github-app/installations/new?state=${organizationId}`
+
+  const dispatch = useDispatch<AppDispatch>()
+  const authProviderLoadingStatus = useSelector<RootState, LoadingStatus>(
+    (state) => getAuthProviderState(state).loadingStatus
+  )
+  const authProviders = useSelector<RootState, GitAuthProvider[]>((state) => selectAllAuthProvider(state))
+  const [githubAuthProvider, setGithubAuthProvider] = useState<GitAuthProvider>()
+
+  const repositories = useSelector<RootState, RepositoryEntity[]>((state) => selectAllRepository(state))
+  const repositoriesLoadingStatus = useSelector<RootState, LoadingStatus>(
+    (state) => getRepositoryState(state).loadingStatus
+  )
+
+  const { getAccessTokenSilently } = useAuth()
+  const { openModal, closeModal } = useModal()
+
+  useEffect(() => {
+    getAccessTokenSilently({
+      ignoreCache: true,
+    }).then(() => {
+      dispatch(fetchAuthProvider({ organizationId }))
+    })
+  }, [getAccessTokenSilently, dispatch, organizationId])
+
+  useEffect(() => {
+    authProviders.forEach((authProvider) => {
+      authProvider.name === 'GITHUB' && setGithubAuthProvider(authProvider)
+    })
+  }, [authProviders])
+
+  useEffect(() => {
+    if (githubAuthProvider?.use_bot) {
+      dispatch(fetchRepository({ organizationId: organizationId, gitProvider: GitProviderEnum.GITHUB }))
+    }
+  }, [githubAuthProvider, dispatch, organizationId])
+
+  const onConfigure = () => {
+    window.open(githubConnectUrl, '_blank')
+  }
+
+  const onDisconnectWithModal = () => {
+    openModal({
+      content: <DisconnectionConfirmModal onClose={closeModal} onSubmit={() => onDisconnect(true)} />,
+    })
+  }
+
+  // onDisconnect can be called with force parameter. If it's false, the api call will eventually fail with a 400 error
+  // if the user has some apps that used some repositories of github app. If it fails, we open a modal to confirm the disconnection
+  // and we force the disconnection with force parameter set to true. To put simply: if the user does not use any app
+  // he can disconnect without modal, if he uses some apps, he has to confirm the disconnection
+  const onDisconnect = (force?: boolean) => {
+    dispatch(
+      disconnectGithubApp({
+        organizationId,
+        force: force,
+      })
+    )
+      .unwrap()
+      .then(() => {
+        getAccessTokenSilently({
+          ignoreCache: true,
+        }).then(() => {
+          dispatch(fetchAuthProvider({ organizationId }))
+        })
+        dispatch(repositorySlice.actions.removeAll())
```

Comment:
Seems to be a new implementation, why you didn't use an addCase to apply the action and use the removeAll on the disconnectGithubApp dispatch?

Reply:
I'm not sure to understand what you mean by use the removeAll on the disconnectGithupApp dispatch but: 
I did not add a addCase here because addCase are for asyncThunk that triggers asynchronous action, here it's just a basic action that is a direct edition of the store. Also, it's an action directly available in the adapter and other actions like these are added directly in reducers by nx generators
<img width="415" alt="CleanShot 2023-02-20 at 16 53 07@2x" src="https://user-images.githubusercontent.com/6163954/220151923-c76cbfc9-8613-448f-b23d-c4c41b94e7ff.png">

Reply:
I think it's not a good practice to add this one outside the slices, if you need this action after the fetchAuthProvider, you can add an optional param on the fetchAuthProvider like this one: `dispatch(fetchAuthProvider({ organizationId, removeAll }))` and add the action inside the slice

BTW, if it's available in the Nx generators, it's not going to say that it's a good thing. Nx puts forward examples and not things to use imperatively like this

Reply:
I really don't see why it's a bad practice, please provide more arguments or indicate me a documentation or an article that goes in this direction so I can make my mind up.

Furthermore, `authProviderSlice` and `repositorySlice` are two different slices. I can't call the  the method repositoryAdapter.removeAll() from the `authProviderSlice` since the first argument is the state.

So I have to call it via a dispatch where I did it, and creating an `asyncThunk` when you don't have anything async inside does not make sense: I won't be able to return anything from the asyncThunk and there will not be a pending, fullfill and rejected case since there is nothing async here once again.

Reply:
When I talk about bad practices, it's more about a new case, we didn't have this case for another feature, sorry I thought it was easy to understand without an article, I talked about the structure of the project and for this particular case seems to be complicated to find a correct article. But don't worry if you think we didn't have a choice you can do it and just if you can document this one because is very particular 👍